MapEditor/Components/CollisionController.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Drawing;
+using System.Linq;
+using MapEditor.Common;
+
+namespace MapEditor.Components
+{
+    public interface ICollider
+    {
+        Point Position { get; set; }
+        bool IsCollided(ICollider collider);
+    }
+
+    public class BoundingCircle : ICollider
+    {
+        public float Radius { get; set; }
+        public Point Position { get; set; }
+
+        public bool IsCollided(ICollider collider)
+        {
+            var circle = collider as BoundingCircle;
+            if (circle != null)
+            {
+                return IsCollided(circle);
+            }
+
+            var box = collider as BoundingBox;
+            if (box != null)
+            {
+                return IsCollided(box);
+            }
+            return false;
+        }
+
+        private bool IsCollided(BoundingCircle collider)
+        {
+            var distanceX = Position.X - collider.Position.X;
+            var distanceY = Position.Y - collider.Position.Y;
+
+            var magnitudeSquared = distanceX * distanceX + distanceY * distanceY;
+            return magnitudeSquared < (Radius + collider.Radius) * (Radius + collider.Radius);
+        }
+
+        private bool IsCollided(BoundingBox box)
```

Comment:
Think about how these three situations can be make more generic for code sharing

Reply:
Done via extension method